frontend/src/components/cardInfo/cardInfo.js

```diff
@@ -10,19 +10,86 @@ import Button from "../button/button";
 import { useNavigate } from "react-router";
 import { FRONTEND_URL } from "../../utils/constants";
 import addressService from "../../services/address.service";
+import validation from "../../utils/validation";
+import { STATUS_CODES } from "../../utils/httpStatusCode";
+import productService from "../../services/product.service";
+import personService from "../../services/person.service";
+import pictureService from "../../services/picture.service";
 
-function CardInfo({ props }) {
+function CardInfo({ props, location, setError }) {
   const elements = useElements();
   const stripe = useStripe();
   const [validCard, setValidCard] = useState();
   const navigate = useNavigate();
   const [paymentError, setPaymentError] = useState();
+  const seller = personService.getCurrentUser();
+  const [address, setAddress] = useState();
+
+  function saveCard(newProductId) {
+    if (!stripe || !elements) {
+      return;
+    }
+
+    stripe
+      .confirmSetup({
+        elements,
+        confirmParams: {
+          return_url: `${FRONTEND_URL}/product/${newProductId}`,
+        },
+      })
+      .then((result) => {
+        if (result.error) setPaymentError(result.error.message);
+      });
+  }
+
+  async function savePhotos(photo) {
```

Comment:
Can we make a method which sends list of images and saves it ? 
With that, we can spare frontend of looping and sending too many requests on cloudinary (number of pictures = number of requests)